includes/admin_import.php

```diff
@@ -4,743 +4,1212 @@
+				tsml_delete(tsml_get_data_source_ids($_POST['tsml_remove_data_source']));
 
-			//clean up orphaned locations & groups
-			tsml_delete_orphans();
+				//clean up orphaned locations & groups
+				tsml_delete_orphans();
 
-			//remove data source
-			unset($tsml_data_sources[$_POST['tsml_remove_data_source']]);
-			update_option('tsml_data_sources', $tsml_data_sources);
+				//remove data source
+				unset($tsml_data_sources[$_POST['tsml_remove_data_source']]);
+				update_option('tsml_data_sources', $tsml_data_sources);
 
-			tsml_alert(__('Data source removed.', '12-step-meeting-list'));
+				tsml_alert(__('Data source removed.', '12-step-meeting-list'));
+			}
 		}
-	}
-
-	//change program
-	if (!empty($_POST['tsml_program']) && isset($_POST['tsml_nonce']) && wp_verify_nonce($_POST['tsml_nonce'], $tsml_nonce)) {
-		$tsml_program = sanitize_text_field($_POST['tsml_program']);
-		update_option('tsml_program', $tsml_program);
-		tsml_alert(__('Program setting updated.', '12-step-meeting-list'));
-	}
 
-	//change distance units
-	if (!empty($_POST['tsml_distance_units']) && isset($_POST['tsml_nonce']) && wp_verify_nonce($_POST['tsml_nonce'], $tsml_nonce)) {
-		$tsml_distance_units = ($_POST['tsml_distance_units'] == 'mi') ? 'mi' : 'km';
-		update_option('tsml_distance_units', $tsml_distance_units);
-		tsml_alert(__('Distance units updated.', '12-step-meeting-list'));
-	}
+		//change program
+		if (!empty($_POST['tsml_program']) && isset($_POST['tsml_nonce']) && wp_verify_nonce($_POST['tsml_nonce'], $tsml_nonce)) {
+			$tsml_program = sanitize_text_field($_POST['tsml_program']);
+			update_option('tsml_program', $tsml_program);
+			tsml_alert(__('Program setting updated.', '12-step-meeting-list'));
+		}
 
-	//change distance units
-	if (!empty($_POST['tsml_contact_display']) && isset($_POST['tsml_nonce']) && wp_verify_nonce($_POST['tsml_nonce'], $tsml_nonce)) {
-		$tsml_contact_display = ($_POST['tsml_contact_display'] == 'public') ? 'public' : 'private';
-		update_option('tsml_contact_display', $tsml_contact_display);
-		tsml_cache_rebuild(); //this value affects what's in the cache
-		tsml_alert(__('Contact privacy updated.', '12-step-meeting-list'));
-	}
+		//change distance units
+		if (!empty($_POST['tsml_distance_units']) && isset($_POST['tsml_nonce']) && wp_verify_nonce($_POST['tsml_nonce'], $tsml_nonce)) {
+			$tsml_distance_units = ($_POST['tsml_distance_units'] == 'mi') ? 'mi' : 'km';
+			update_option('tsml_distance_units', $tsml_distance_units);
+			tsml_alert(__('Distance units updated.', '12-step-meeting-list'));
+		}
 
-	//change sharing setting
-	if (!empty($_POST['tsml_sharing']) && isset($_POST['tsml_nonce']) && wp_verify_nonce($_POST['tsml_nonce'], $tsml_nonce)) {
-		$tsml_sharing = ($_POST['tsml_sharing'] == 'open') ? 'open' : 'restricted';
-		update_option('tsml_sharing', $tsml_sharing);
-		tsml_alert(__('Sharing setting updated.', '12-step-meeting-list'));
-	}
+		//change distance units
```

Comment:
this should read "change contact display" -- it was wrong before but would be great to fix it here